core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java

```diff
@@ -1072,7 +1097,22 @@ public static void deleteLoadsAndUpdateMetadata(CarbonTable carbonTable, boolean
                 isUpdateRequired(isForceDeletion, carbonTable,
                     identifier, details, cleanStaleInprogress);
             if (!tuple2.isUpdateRequired) {
-              return;
+              try {
+                for (LoadMetadataDetails oneLoad : details) {
+                  if (isExpiredSegment(oneLoad, carbonTable.getAbsoluteTableIdentifier())) {
+                    if (!carbonTable.isHivePartitionTable()) {
+                      trashSizeRemaining += FileFactory.getDirectorySize(CarbonTablePath
+                        .getSegmentPath(carbonTable.getTablePath(), oneLoad.getLoadName()));
+                    } else {
+                      trashSizeRemaining += partitionTableSegmentSize(carbonTable, oneLoad,
+                        details, partitionSpecs);
+                    }
+                  }
+                }
+              } catch (Exception e) {
+                LOG.error("Unable to calculate size of garbage data", e);
+              }
+              return new long[]{sizeFreed, trashSizeRemaining};
```

Comment:
dry run is meant to give the size, do you need to give empty size or fail it? if you give empty its purpose itself not completed.

Reply:
we will just print the exception and carry on with the clean files command, not failing the whole thing, in case there is some error while calculating  the size, i guess it will return 0 then, or in case when nothing is deleted from clean files operation, then also we sill return the size freed as 0. Can have a discussion on this and change accrodingly.

Reply:
when nothing is freed by clean files, returning 0 is ok. But when some exception happens in dry run may be better to throw exception

Reply:
yes, agree with @ajantha-bhat , i meant the same

Reply:
changed code, dry run is separate from the current clean files code and not mixed with the existing code